### ISSUE TITLE
Add classifiers to twisted.web setup.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6695]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6695|
|Type|enhancement|
|Created|2013-08-23 17:59:28Z|
|Branch|https://github.com/twisted/twisted/tree/web-classifiers-6695|

The `dist.setup` call in [twisted/web/topfiles/setup.py](twisted/web/topfiles/setup.py) accepts an `extraMeta` dict where [classifiers](https://pypi.python.org/pypi?%3Aaction=list_classifiers) can be defined. This is currently missing for twisted.web.

Attachments:

* [classifiers-6695.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/00f/00f27118b84eeb1fd935443a8f8c2158d5d8b93d/9267d3c7689d44fcfc754feada85e44ee8740f10.patch) (993 bytes) - added by hawkowl on 2013-08-28 17:25:59Z - Patch that adds classifiers to t.w.topfiles.setup
* [classifiers-6695-v2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/00f/00f27118b84eeb1fd935443a8f8c2158d5d8b93d/74ca70f2ea6096aeb1faebd32ec9f0986e4e446f.patch) (847 bytes) - added by hawkowl on 2013-08-28 17:35:03Z - Patch that adds classifiers to t.w.topfiles.setup - v2, with correct ordering

<details><summary>Searchable metadata</summary>

```
trac-id__6695 6695
type__enhancement enhancement
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_web_classifiers_6695 branches/web-classifiers-6695
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1377280768000000 1377280768000000
changetime__1378940515000000 1378940515000000
version__None None
owner__thijs thijs
cc__thijs cc__jknight cc__hawkowl@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

I've gone and added the classifiers in.

The extraMeta dict seemed to be there so that the classifiers could be defined for >2.4, but since the merge that removed 2.4isms, it can be put inline, which is what this patch has done. I had also put classifiers before long_description, but after uploading, I tripped over dist.py which had it after long_description, which v2 fixes, so that it's all uniform.

The classifiers in the patch are as follows - they look correct to me, but I'm not sure if I've missed/misclassified anything.

```
Development Status :: 5 - Production/Stable
Environment :: No Input/Output (Daemon)
Intended Audience :: Developers
License :: OSI Approved :: MIT License
Programming Language :: Python
Topic :: Internet :: WWW/HTTP :: HTTP Servers
Topic :: Internet :: WWW/HTTP :: WSGI
Topic :: Software Development :: Libraries :: Python Modules
```

I've confirmed it works via running `python twisted/web/topfiles/setup.py --classifiers`. I don't know how one would write a test for this, though.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

Thanks for your patch, I'll put it in a branch and if buildbot is happy, merge it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [39955]) Branching to 'web-classifiers-6695'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [39956]) apply classifiers-6695-v2.patch, refs [#6695](https://github.com/twisted/twisted/issues/6695)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/web-classifiers-6695).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [39957]) Merge web-classifiers-6695: Add pypi classifiers to twisted.web.

Author: hawkowl
Reviewer: thijs
Fixes: [#6695](https://github.com/twisted/twisted/issues/6695)
